sdk/storage/Azure.Storage.Blobs/tests/BlobClientTests.cs

```diff
@@ -872,26 +872,44 @@ public async Task ChecksForCancelation()
         {
             await using DisposingContainer test = await GetTestContainerAsync();
 
-            // Upload a GB
+            // Upload 100MB
             var data = GetRandomBuffer(100 * Constants.MB);
             BlobClient blob = InstrumentClient(test.Container.GetBlobClient(GetNewBlobName()));
             using var stream = new MemoryStream(data);
             await blob.UploadAsync(stream);
 
-            // Create a CancellationToken that times out after .01s
-            CancellationTokenSource source = new CancellationTokenSource(TimeSpan.FromSeconds(.01));
-            CancellationToken token = source.Token;
+            await AssertDownloadAsync();
+            await AssertDownloadToAsync();
 
-            // Verifying downloading will cancel
-            try
+            async Task AssertDownloadAsync()
             {
-                using BlobDownloadInfo result = await blob.DownloadAsync(cancellationToken: token);
+                // Create a CancellationToken that times out after .01s
+                CancellationTokenSource source = new CancellationTokenSource(TimeSpan.FromSeconds(.01));
+                CancellationToken token = source.Token;
+                // Delay 1s to ensure that the Download operation should be canceled since DownloadAsync doesn't
+                // buffer the response.
+                await Delay(1000);
```

Comment:
You've changed how the test works by doing this. Before the method started with a non-cancelled token and it was cancelled during the download, now you're always passing a cancelled token in.

Reply:
The DownloadTo call uses a non-cancelled token. I couldn't get the Download test to reliably pass since the initial response might take a few milliseconds.

Reply:
Also, I don't think passing the cancelled token should change the coverage as far as our library goes, since the token checking that should cause the task to be cancelled would still occur outside of Storage (in HttpClient I believe).